core/vm/evm.go

```diff
@@ -58,22 +58,12 @@ func (evm *EVM) precompile(addr common.Address) (PrecompiledContract, bool) {
 	return p, ok
 }
 
-// run runs the given contract and takes care of running precompiles with a fallback to the byte code interpreter.
+// run runs the given contract and takes care of running precompiles
 func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, error) {
-	for _, interpreter := range evm.interpreters {
-		if interpreter.CanRun(contract.Code) {
-			if evm.interpreter != interpreter {
-				// Ensure that the interpreter pointer is set back
-				// to its current value upon return.
-				defer func(i Interpreter) {
-					evm.interpreter = i
-				}(evm.interpreter)
-				evm.interpreter = interpreter
-			}
-			return interpreter.Run(contract, input, readOnly)
-		}
+	if !evm.interpreter.CanRun(contract.Code) {
```

Comment:
That `CanRun` function has been introduced to check that it was WASM code. If we no longer support multiple interpreters, then it should go as well, since in EVM, any bytecode is considered valid.

Reply:
I went ahead and deleted the `Interpreter` interface and the `run` func defined in evm.go too (in addition to `CanRun`). But happy to revert these changes and only remove `CanRun`